### PR TITLE
#11721: Ignore .hypothesis

### LINKED ISSUE #11721: .hypothesis should be in .gitignore
**Describe the incorrect behavior you saw**

`git diff` shows me an untracked `.hypothesis` directory.

**Describe how to cause this behavior**

I ran the tests.

**Describe the correct behavior you'd like to see**

`.hypothesis` [shouldn't be checked in](https://hypothesis.readthedocs.io/en/latest/database.html#the-hypothesis-example-database), so it should be ignored by `git`.

### PR DESCRIPTION
**Pull request by twm**
## Scope and purpose

Fixes #11721. See there for an exhaustive discussion.

### CODE REVIEW

**Comment by twm** [comment]
please review

File: .github/scripts/check-pr-text.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def fail(message):
 if not pr_body:
     fail("Body for the PR not found. " "Maybe missing PR_BODY env var.")
 
-title_search = re.search(r"^(#\d+) .+", pr_title)
+title_search = re.search(r"^(#\d+):? .+", pr_title)
```
Comment (adiroiban) [inline .github/scripts/check-pr-text.py:44]: I think that we should also update the PR template to document that multiple title formats are allowed.

AFAIK, the PR template is the only documenting for our PR  naming convention.

**Review by adiroiban — APPROVED** [review]
Thanks. All good. 
Only a small change regarding the drive-by change.

File: .github/scripts/check-pr-text.py (hunk 1 above)
Comment (twm) [reply .github/scripts/check-pr-text.py:44]: :shrug: I think that a bot should be maintaining this invariant, not human beings.

**Pull request merged**